code/modules/shuttle/spaceship_navigation_beacon.dm

```diff
@@ -0,0 +1,63 @@
+/obj/item/circuitboard/machine/spaceship_navigation_beacon
+	name = "Bluespace Navigation Gigabeacon (Machine Board)"
+	build_path = /obj/machinery/spaceship_navigation_beacon
+	req_components = list()
+
+
+/obj/machinery/spaceship_navigation_beacon
+	name = "Bluespace Navigation Gigabeacon"
+	desc = "A device that creates a bluespace anchor that allow ships jump near to it."
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "core"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 0
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/spaceship_navigation_beacon
+	
+	var/locked = FALSE //Locked beacons don't allow to jump to it.
+
+
+/obj/machinery/spaceship_navigation_beacon/Initialize()
+	. = ..()
+	SSshuttle.beacons |= src
+
+obj/machinery/spaceship_navigation_beacon/emp_act()
+	locked = TRUE
+
+/obj/machinery/spaceship_navigation_beacon/Destroy()
+	SSshuttle.beacons -= src
+	return ..()
+
+// update the icon_state
+/obj/machinery/spaceship_navigation_beacon/update_icon()
+	if(powered())
+		icon_state = "core"
+	else
+		icon_state = "core-open"
+
+/obj/machinery/spaceship_navigation_beacon/power_change()
+	. = ..()
+	update_icon()
+
+/obj/machinery/spaceship_navigation_beacon/multitool_act(mob/living/user, obj/item/multitool/I)
+	if(panel_open)
+		var/new_name = "Beacon_[input("Enter the custom name for this beacon", "It be Beacon ..your input..") as text]"
```

Comment:
What exactly is the second string for? 
`"It be beacon ..your input.."`

Reply:
It's beckoning your input.
![rename](https://user-images.githubusercontent.com/31044876/68547507-35e25280-03da-11ea-97f5-796acb097d89.PNG)